File: oss/src/main/java/com/truenine/component/oss/amazon/S3Policies.java
package com.truenine.component.oss.amazon;

public class S3Policies {

  public static class Effect {
    public static final String ALLOW = "Allow";
  }

  public static class Bucket {
    public static final String GET_LOCATION = "GetBucketLocation";
    public static final String LIST = "ListBucket";
    public static final String LIST_MUL_UPLOADS = "ListBucketMultipartUploads";
  }

  public static class Obj {
    public static final String GET = "GetObject";
    public static final String PUT = "PutObject";
    public static final String DEL = "DeleteObject";
    public static final String LIST_MUL_UPLOAD_PARTS = "ListMultipartUploadParts";
    public static final String ABORT_MUL_UPLOAD = "AbortMultipartUpload";
  }
}

File: oss/shared/src/main/kotlin/io/github/truenine/composeserver/oss/S3Policies.kt
package io.github.truenine.composeserver.oss

interface S3Policies {
  interface Effect {
    companion object {
      const val ALLOW: String = "Allow"
    }
  }

  interface Bucket {
    companion object {
      const val GET_LOCATION: String = "GetBucketLocation"
      const val LIST: String = "ListBucket"
      const val LIST_MUL_UPLOADS: String = "ListBucketMultipartUploads"
    }
  }

  interface Object {
    companion object {
      const val GET: String = "GetObject"
      const val PUT: String = "PutObject"
      const val DEL: String = "DeleteObject"
      const val LIST_MUL_UPLOAD_PARTS: String = "ListMultipartUploadParts"
      const val ABORT_MUL_UPLOAD: String = "AbortMultipartUpload"
    }
  }
}
